lib/modules/manager/gradle/parser/common.ts

```diff
@@ -0,0 +1,101 @@
+import type { lexer } from 'good-enough-parser';
+import type { Ctx, NonEmptyArray, PackageVariables } from '../types';
+
+export const REGISTRY_URLS = {
+  google: 'https://dl.google.com/android/maven2/',
+  gradlePluginPortal: 'https://plugins.gradle.org/m2/',
+  jcenter: 'https://jcenter.bintray.com/',
+  mavenCentral: 'https://repo.maven.apache.org/maven2',
+};
+
+export const ANNOYING_METHODS: ReadonlySet<string> = new Set([
+  'createXmlValueRemover',
+  'events',
+  'args',
+  'arrayOf',
+  'listOf',
+  'mutableListOf',
+  'setOf',
+  'mutableSetOf',
+  'stages', // https://github.com/ajoberstar/reckon,
+  'mapScalar', // https://github.com/apollographql/apollo-kotlin
+]);
+
+export function storeVarToken(ctx: Ctx, node: lexer.Token): Ctx {
```

Comment:
it seems `Ctx` should be refactored to a class, as those are all candidates for class methods. Can be done in a follow up pr.

Reply:
Context is meant to be plain data and is subject to internal deep-cloning